tests/i3c_tests.py

```diff
@@ -0,0 +1,311 @@
+        # This test assumes that the are no devices connected to the bus.
+
+        if self.use_simulator:
+            self.skipTest("For real device only")
+
+        (success, result) = self.i3c.init_bus(1500)
+
+        self.assertEqual(success, False, "Init bus succeded, are there devices connected?")
+        self.assertTrue("errors" in result)
+
+    def test_i3c_reset_bus(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        self.i3c.init_bus(2000)
+        (success, result) = self.i3c.reset_bus()
+
+        self.assertTupleEqual((success, result), (True, 2000))
+
+    def test_i3c_reset_bus_before_init(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        (success, _) = self.i3c.reset_bus()
+
+        self.assertEqual(success, True)
+
+    def test_i3c_targets(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        self.i3c.init_bus(3300)
+        (success, targets) = self.i3c.targets()
+
+        self.assertEqual(success, True)
+        self.assertEqual(len(targets), 3)
+        self.assertDictEqual(targets[0], {
+            "static_address": 0x50,
+            "dynamic_address": 0x08,
+            "bcr": 0x10,
+            "dcr": 0xC3,
+            "pid": ["0x00", "0x00", "0x00", "0x00", "0x64", "0x65"],
+        })
+        self.assertDictEqual(targets[1], {
+            "static_address": 0x51,
+            "dynamic_address": 0x09,
+            "bcr": 0x03,
+            "dcr": 0x63,
+            "pid": ["0x02", "0x17", "0x0F", "0x1D", "0x00", "0x5A"],
+        })
+        self.assertDictEqual(targets[2], {
+            "static_address": 0x52,
+            "dynamic_address": 0x0A,
+            "bcr": 0x10,
+            "dcr": 0xC3,
+            "pid": ["0x06", "0x06", "0x06", "0x06", "0x66", "0x66"],
+        })
+
+    def test_i3c_targets_when_bus_not_initialized(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        (success, targets) = self.i3c.targets()
+
+        self.assertTupleEqual((success, targets), (True, []))
+
+    def test_i3c_write_operation_when_bus_not_initialized(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        (success, _) = self.i3c.write(
+            0x08,
+            self.i3c.TransferMode.I3C_SDR,
+            [0x00, 0x00],
+            [0xDE, 0xAD, 0xBE, 0xEF]
+        )
+
+        self.assertEqual(success, False)
+
+    def test_i3c_read_operation_when_bus_not_initialized(self):
+        if not self.use_simulator:
+            self.skipTest("For simulator only")
+
+        (success, _) = self.i3c.read(
+            0x08,
+            self.i3c.TransferMode.I3C_SDR,
+            [0x00, 0x00],
+            1
+        )
+
+        self.assertEqual(success, False)
+
+    def test_i3c_successful_write_operation_on_target_should_return_none(self):
```

Comment:
I know this test was already named like this, but, is not a little bit to specific and long the name of the test? xD